src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -358,17 +358,38 @@ impl NonOverlapSubLevelPicker {
             return vec![];
         }
 
+        let mut expected = Vec::with_capacity(scores.len());
+        let mut unexpected = vec![];
+
+        for selected_task in scores {
+            if selected_task.sstable_infos.len() > MAX_COMPACT_LEVEL_COUNT {
```

Comment:
Shall we do this optimization also for `max_file_number` ?

Reply:
I think file_count is easier to reach the limit than level_count, so we prioritize the level count, and then determine if we need to limit `max_file_number` based on subsequent performance.